### PR TITLE
storage: Remove rdkafka full queue workaround

### PR DESCRIPTION
**Pull request by def-**
Since #24864 has been fixed. via https://buildkite.com/materialize/nightlies/builds/6316#018d7940-69f7-4163-9359-6990c01ed1e0

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by petrosagg** [comment]
Can you rebase this on top of https://github.com/MaterializeInc/materialize/pull/24973 so that you revert the bit of code added by that PR too?

**Comment by def-** [comment]
Done, but why even merge #24973?

File: src/storage/src/sink/kafka.rs [hunk 1]
```diff
@@ -205,6 +205,7 @@ struct TransactionalProducer {
     /// The timeout to use for network operations.
     socket_timeout: Duration,
     /// The maximum duration of a transaction.
+    #[allow(dead_code)]
```
Comment (nrainer-materialize) [inline src/storage/src/sink/kafka.rs:208]: Why is this needed?

**Review by nrainer-materialize — APPROVED** [review]

**Review by nrainer-materialize — APPROVED** [review]

**Comment by petrosagg** [comment]
actually wait, we can't merge this yet. PR https://github.com/MaterializeInc/materialize/pull/24919 which is what fixed the referenced issue did so without upgrading librdkafka in the end, which is required for this code to be removed. So let's just change the comment to say "this logic can be removed once we upgrade to librdkafka 2.3 and increase the queue limits"

**Review by petrosagg — CHANGES_REQUESTED** [review]
requesting changes so that this doesn't get merged accidentally

**Review by petrosagg — APPROVED** [review]

**Review by nrainer-materialize — APPROVED** [review]
